Sources/ClientRuntime/Networking/Http/URI.swift

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+import Foundation
+
+/// Universal Resource Identifier component used to construct target location for a Request.
+public struct URI: Hashable {
```

Comment:
I suggest moving URI to the `Message` directory - it isn't necessarily part of networking (you can have a `file://` uri), and def not just http.